lib/client.js

```diff
@@ -598,18 +597,23 @@ class Client {
             this.error(500, err);
           }
           application.logException(err);
-          return;
         }
         callback();
       }
     };
 
-    try {
-      fn(this, done);
-    } catch (err) {
-      this.error(500, err);
-      application.logException(err);
-      callback();
+    if (fn.constructor.name === 'AsyncFunction') {
```

Comment:
It would be safer to write
```suggestion
    if (Object.prototype.toString.call(fn) === '[object AsyncFunction]') {
```